pkg/scanner/utils/utils.go

```diff
@@ -14,15 +15,34 @@ var (
 )
 
 func MatchVersions(currentVersion *version.Version, rangeVersions []string) bool {
-	for _, p := range rangeVersions {
-		c, err := version.NewConstraint(replacer.Replace(p))
+	v, err := semver.NewVersion(currentVersion.String())
+	if err != nil {
+		log.Logger.Error("NewVersion", "error", err)
+		return false
+	}
+	for i := range rangeVersions {
+		rangeVersions[i] = replacer.Replace(rangeVersions[i])
+		c, err := semver.NewConstraint(rangeVersions[i])
 		if err != nil {
-			log.Logger.Debug("NewConstraint", "error", err)
-			return false
+			log.Logger.Error("NewConstraint", "error", err)
+			continue
 		}
-		if c.Check(currentVersion) {
+		// Validate a version against a constraint.
+		a, msgs := c.Validate(v)
```

Comment:
What is `a`? It is ok if the short name is clear what it means like `for _, m := range msgs {`, but in this case, it is difficult to understand what `a` indicates.

Reply:
Fixed